connect/src/test/scala/org/apache/spark/sql/connect/planner/SparkConnectProtoSuite.scala

```diff
@@ -0,0 +1,60 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.sql.connect.planner
+
+import org.apache.spark.connect.proto
+import org.apache.spark.sql.catalyst.dsl.expressions._
+import org.apache.spark.sql.catalyst.dsl.plans._
+import org.apache.spark.sql.catalyst.expressions.AttributeReference
+import org.apache.spark.sql.catalyst.plans.PlanTest
+import org.apache.spark.sql.catalyst.plans.logical.LocalRelation
+import org.apache.spark.sql.types.{DataType, IntegerType}
+
+/**
+ * This suite is based on connect DSL and test that given same dataframe operations, whether
+ * connect could construct a proto plan that can be translated back, and after analyzed, be the
+ * same as Spark dataframe's generated plan.
+ */
+class SparkConnectProtoSuite extends PlanTest with SparkConnectPlanTest {
+
+  lazy val connectTestRelation = createLocalRelationProto(Map("id" -> IntegerType))
+
+  lazy val sparkTestRelation: LocalRelation = LocalRelation(AttributeReference("id", IntegerType)())
+
+  test("Basic select") {
+    val connectPlan = {
+      // TODO: Scala only allows one implicit per scope so we keep proto implicit imports in
+      // this scope. Need to find a better way to make two implicits work in the smae scope.
+      import org.apache.spark.sql.catalyst.connect.expressions._
+      import org.apache.spark.sql.catalyst.connect.plans._
+      transform(connectTestRelation.select("id".protoAttr))
+    }
+    val sparkPlan = sparkTestRelation.select("id")
+    comparePlans(connectPlan.analyze, sparkPlan.analyze, false)
+  }
+
+  private def createLocalRelationProto(colNameTypeMap: Map[String, DataType]): proto.Relation = {
```

Comment:
it's weird to pass a `Map` here, as the elements order is not deterministic. We should use `Seq[(String, DataType)]` or simply `Seq[AttributeReference]`

Reply:
Good point on the ordering.

I changed to use `Seq[(String, DataType)]`. I guess eventually I will change it to `Seq[AttributeReference]` when we are ready to test all the fields of a `AttributeReference`. Right now only String is really used due to no data type support in proto.

Reply:
Seq[AttributeReference] is better actually based on your comment below.